### ISSUE TITLE
Log collection using system-tools fails with fatal error for a RKE cluster created in Rancher

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#16609]
Version : system-tools version v0.1.1-rc1

Steps:
1. Create a RKE cluster using DO template in Rancher(v2.1.2-rc12)
2. Save the kubeconfig for this cluster as a file
3. Run the command  
```
./system-tools_darwin-amd64 logs --kubeconfig config.yml 
```
It fails with the below error:
```
soumyas-MacBook-Pro:systemtools soumya$ ./system-tools_darwin-amd64 logs --kubeconfig config.yml 
INFO[0000] deploying log collection DaemonSet [log-collector].. 
INFO[0000] wating for DaemonSet [log-collector] to be ready.. 
INFO[0000] log collection DaemonSet [log-collector] deployed successfully.. 
INFO[0000] starting log collection..                    
INFO[0000] fetching logs from node [soumyasystooltest1].. 
INFO[0000] removing log collection DaemonSet [log-collector].. 
INFO[0001] log collection DaemonSet [log-collector] removed successfully.. 
FATA[0001] error executing command on pod [cattle-system/log-collector-99kff]: unable to upgrade connection: container not found ("log-collector") 
```

Note: The same errors were seen for custom and imported rke clusters

**Comment by alena1108** [comment]
The fix is available in v0.1.1-rc2

**Comment by soumyalj** [comment]
Tested with system-tools version v0.1.1-rc2 , log collection works, but logs for only one node is available. The logs for all nodes is fetched, but in the tar file, logs for only one node is available.
```
soumyas-MacBook-Pro:systemtools soumya$ ./system-tools_darwin-amd64test logs --kubeconfig kubeconfig.yml
INFO[0000] deploying log collection DaemonSet [log-collector].. 
INFO[0000] wating for DaemonSet [log-collector] to be ready.. 
INFO[0002] log collection DaemonSet [log-collector] deployed successfully.. 
INFO[0003] starting log collection..                    
INFO[0003] fetching logs from node [soumyadobugtesting1].. 
INFO[0004] fetching logs from node [soumyadobugtesting2].. 
INFO[0004] removing log collection DaemonSet [log-collector].. 
INFO[0004] log collection DaemonSet [log-collector] removed successfully.. 

```
When the tar file is extracted, we can see logs for only one node.
```
soumyas-MacBook-Pro:systemtools soumya$ tar -xvf cluster-logs.tar 
x soumyadobugtesting1/
x soumyadobugtesting1/kube-scheduler.log
x soumyadobugtesting1/kubelet.log
x soumyadobugtesting1/kube-apiserver.log
x soumyadobugtesting1/etcd.log
x soumyadobugtesting1/kube-proxy.log
x soumyadobugtesting1/kube-controller-manager.log

```
Logs for all nodes should be present in the tar file.

**Comment by alena1108** [comment]
The fix should be available in v0.1.1-rc3

**Comment by soumyalj** [comment]
Tested with system-tools version v0.1.1-rc3. 
Steps:

1. Create a 3 node RKE cluster using DO template in Rancher master
2. Save the kubeconfig for this cluster as a file
3. Run the command

```
./system-tools_darwin-amd64 logs --kubeconfig config.yml 
```
Log collection works, the tar file generated contains logs for all the 3 nodes. Each of the nodes have the following logs :
1. etcd.log	
2. kube-apiserver.log		
3. kube-controller-manager.log
4. kube-proxy.log
5. kube-scheduler.log
6. kubelet.log
